tests/helpers/test_entity.py

```diff
@@ -114,83 +114,76 @@ def test_device_class(self):
 
-    with patch.object(hass.loop, "call_later") as mock_call:
-        await mock_entity.async_update_ha_state(True)
-        assert mock_call.called
-        assert len(mock_call.mock_calls) == 2
-
-        timeout, logger_method = mock_call.mock_calls[0][1][:2]
-
-        assert timeout == entity.SLOW_UPDATE_WARNING
-        assert logger_method == entity._LOGGER.warning
-
-        assert mock_call().cancel.called
+    fast_update_time = 0.0000001
 
+    with patch.object(entity, "SLOW_UPDATE_WARNING", fast_update_time):
+        await mock_entity.async_update_ha_state(True)
+        assert str(fast_update_time) in caplog.text
+        assert mock_entity.entity_id in caplog.text
         assert update_call
 
 
-async def test_warn_slow_update_with_exception(hass):
+async def test_warn_slow_update_with_exception(hass, caplog):
     """Warn we log when entity update takes a long time and trow exception."""
     update_call = False
 
     async def async_update():
         """Mock async update."""
         nonlocal update_call
         update_call = True
+        await asyncio.sleep(0.00001)
         raise AssertionError("Fake update error")
 
     mock_entity = entity.Entity()
     mock_entity.hass = hass
     mock_entity.entity_id = "comp_test.test_entity"
     mock_entity.async_update = async_update
 
-    with patch.object(hass.loop, "call_later") as mock_call:
-        await mock_entity.async_update_ha_state(True)
-        assert mock_call.called
-        assert len(mock_call.mock_calls) == 2
-
-        timeout, logger_method = mock_call.mock_calls[0][1][:2]
-
-        assert timeout == entity.SLOW_UPDATE_WARNING
-        assert logger_method == entity._LOGGER.warning
-
-        assert mock_call().cancel.called
+    fast_update_time = 0.0000001
 
+    with patch.object(entity, "SLOW_UPDATE_WARNING", fast_update_time):
+        await mock_entity.async_update_ha_state(True)
+        assert str(fast_update_time) in caplog.text
+        assert mock_entity.entity_id in caplog.text
         assert update_call
 
 
-async def test_warn_slow_device_update_disabled(hass):
+async def test_warn_slow_device_update_disabled(hass, caplog):
     """Disable slow update warning with async_device_update."""
     update_call = False
 
     async def async_update():
         """Mock async update."""
         nonlocal update_call
+        await asyncio.sleep(0.00001)
         update_call = True
 
     mock_entity = entity.Entity()
     mock_entity.hass = hass
     mock_entity.entity_id = "comp_test.test_entity"
     mock_entity.async_update = async_update
 
-    with patch.object(hass.loop, "call_later") as mock_call:
-        await mock_entity.async_device_update(warning=False)
+    fast_update_time = 0.0000001
 
-        assert not mock_call.called
+    with patch.object(entity, "SLOW_UPDATE_WARNING", fast_update_time):
+        await mock_entity.async_device_update(warning=False)
+        assert str(fast_update_time) not in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)